src/main/java/domain/LadderGame.java

```diff
@@ -0,0 +1,31 @@
+package domain;
+
+import java.util.LinkedHashMap;
+import java.util.Map;
+
+public class LadderGame {
+
+    private final Ladder ladder;
+    private final Results unsortedResults;
+    private final Map<Player, Result> playersAndResults;
+
+    public LadderGame(Ladder ladder, Results unsortedResults) {
+        this.ladder = ladder;
+        this.unsortedResults = unsortedResults;
+        this.playersAndResults = new LinkedHashMap<>();
+    }
+    public Result getResultOf(Player player) {
+        if (isResultAlreadyExist(player)) {
+            return playersAndResults.get(player);
+        }
+        Position resultPosition = ladder.getResultPositionOf(player.position());
+        Result result = unsortedResults.get(resultPosition.value());
+        playersAndResults.put(player, result);
+        return result;
+    }
```

Comment:
Map에서 지원하는 메서드로 표현해볼 수 있습니다 😄 
(요런것이 있다 정도를 알려드리려고 예시 드립니다!)
```suggestion
    public Result getResultOf(Player player) {
        return playersAndResults.computeIfAbsent(player, key -> {
            Position resultPosition = ladder.getResultPositionOf(player.position());
            return unsortedResults.get(resultPosition.value());
        });
    }
```

관련 아티클 공유드립니다
[자바8 HashMap 보다 간결하고 효과적으로 작성하기](https://blog.advenoh.pe.kr/java/%EC%9E%90%EB%B0%948-HashMap-%EB%B3%B4%EB%8B%A4-%EA%B0%84%EA%B2%B0%ED%95%98%EA%B3%A0-%ED%9A%A8%EA%B3%BC%EC%A0%81%EC%9C%BC%EB%A1%9C-%EC%9E%91%EC%84%B1%ED%95%98%EA%B8%B0/)

Reply:
헉 이미 잘 만들어진 바퀴가 있었군요 감사합니다!! 🙇